### PR TITLE
add option to keep/remove all intermediate feature lists in batch mode

### PR DESCRIPTION
**Pull request by SteffenHeu**
grafik

### CODE REVIEW

File: src/main/java/io/github/mzmine/modules/batchmode/BatchComponent.fxml [hunk 1]
```diff
@@ -41,80 +40,110 @@
...
+      <Button fx:id="btnRemoveModule" mnemonicParsing="false" onAction="#onRemoveModulePressed"
+        text="&lt;&lt;"/>
+      <GridPane.margin>
+        <Insets/>
+      </GridPane.margin>
+      <padding>
+        <Insets top="5.0"/>
+      </padding>
+    </VBox>
+    <FlowPane alignment="CENTER_LEFT" prefHeight="200.0" prefWidth="200.0" GridPane.columnIndex="2"
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/batchmode/BatchComponent.fxml:117]: need to be defined? or can it just be the calculated size

File: src/main/java/io/github/mzmine/modules/batchmode/BatchComponent.fxml [hunk 2]
```diff
@@ -41,80 +40,110 @@
...
+      GridPane.rowIndex="4">
+      <children>
+        <Button fx:id="btnSetHandleIntermediateFlists" mnemonicParsing="false"
+          onAction="#onHandleIntermediateFlistsPressed" text="Set"/>
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/batchmode/BatchComponent.fxml:121]: Maybe Apply to batch

File: src/main/java/io/github/mzmine/modules/batchmode/BatchComponent.fxml [hunk 3]
```diff
@@ -41,80 +40,110 @@
...
+      </children>
+    </FlowPane>
+    <FlowPane alignment="CENTER_RIGHT" GridPane.halignment="RIGHT" GridPane.rowIndex="4"
+      GridPane.valignment="CENTER">
+      <children>
+        <Label alignment="CENTER" text="Intermediate feature lists"/>
+      </children>
+    </FlowPane>
+    <ComboBox fx:id="cmbHandleFlists" prefWidth="150.0" GridPane.columnIndex="1"
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/batchmode/BatchComponent.fxml:130]: maybe remove prefWidth - should be calculated

File: src/main/java/io/github/mzmine/modules/batchmode/BatchComponent.fxml (hunk 1 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/batchmode/BatchComponent.fxml:117]: that's just how it comes out of the scene builder

File: src/main/java/io/github/mzmine/modules/batchmode/BatchComponentController.java [hunk 4]
```diff
@@ -501,4 +511,31 @@ private TreeItem<Object> cloneTreeItem(TreeItem<Object> item) {
...
+  void onHandleIntermediateFlistsPressed() {
+    final OriginalFeatureListOption option = cmbHandleFlists.getValue();
+    setHandleOriginalFeatureLists(option);
+    var str = switch (option) {
+      case KEEP -> "keep intermediate feature lists.";
+      case REMOVE -> "remove intermediate feature lists.";
+      case PROCESS_IN_PLACE -> "process on feature lists if possible and remove otherwise.";
+    };
+    MZmineCore.getDesktop()
+        .displayMessage("Batch mode updated", "Updated all batch steps to " + str);
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/batchmode/BatchComponentController.java:525]: currently this does not log the message. Maybe we should in general also log those messages?
Or just log it here as well

**Review by robinschmid — APPROVED** [review]
Finally... was waiting for this feature for so long

**Comment by SteffenHeu** [comment]
> Each module already has its own parameter "keep previous feature list?", doesn't it? Doesn't this make it redundant?

yes, but when you have a huge batch you dont want to update all options individually. E.g. when coming from an optimisation, you usually have all options on "KEEP". Now you can set all to REMOVE

**Comment by SteffenHeu** [comment]
It just updates.

**Comment by robinschmid** [comment]
Makes total sense.... This replaces all the values in all modules. We changed it to a specific parameter type so that we can do this.
The idea is that during optimization you want to keep all lists and then switch to the optimized version that processes in place or removes old lists

**Pull request merged**